### PR TITLE
CHANGES: update changelog for 24.0 release

### PR DESCRIPTION
**Pull request by Emantor**
**Description**
 Update the changelog for 24.0.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -10,8 +10,37 @@ New Features in 24.0
...
+- Support for sigrok DMMs was added.
+- Support for Digital Outputs switched via HTTP was added.
+- The ``QEMUDriver`` has a new get_qemu_base_args() function which can be used to
+  extract the arguments passed to qemu.
+- The ``SSHDriver`` has gained support to forward unix sockets.
+- The exporter has gained an ``--fqdn`` argument to set the hostname to the
+  fully qualified domain name instead of the hostname.
+- The ``QEMUDriver`` now has an additional ``disk_opts`` property which can be
   used to pass additional options for the disk directly to QEMU
+- All drivers now inherit a logger from the ``Driver`` baseclass and many
```
Comment (ejoerns) [inline CHANGES.rst:36]:
```suggestion
- All drivers now inherit a logger from the ``Driver`` base class and many
```

File: CHANGES.rst [hunk 2]
```diff
@@ -10,8 +10,37 @@ New Features in 24.0
...
+  drivers were changed to use this logger.
+- The new ``poe_mib`` backend allows switching of power over ethernet capable
```
Comment (ejoerns) [inline CHANGES.rst:38]:
```suggestion
- The new ``poe_mib`` backend allows switching of power over Ethernet-capable
```

File: CHANGES.rst [hunk 3]
```diff
@@ -10,8 +10,37 @@ New Features in 24.0
...
+  ports on switches that use the corresponding SNMP MIB.
+- The ``RawNetworkInterfaceDriver`` allows the replay and recording of network
+  packets on ethernet interfaces.
+- The i.MX93 usb loader ID has been added.
```
Comment (ejoerns) [inline CHANGES.rst:42]:
```suggestion
- The i.MX93 usb loader USB ID has been added.
```

File: CHANGES.rst [hunk 4]
```diff
@@ -22,19 +51,49 @@ Bug fixes in 24.0
   of pip.
 - Several tests have gained an importorskip() call to skip them if the
   module is not available.
+- labgrid now uses it's own pyserial fork from pypi since installation from
```
Comment (ejoerns) [inline CHANGES.rst:54]:
```suggestion
- labgrid now uses its own pyserial fork from pypi since installation from
```

File: CHANGES.rst [hunk 5]
```diff
@@ -22,19 +51,49 @@ Bug fixes in 24.0
...
+  github as an egg is no longer properly supported.
 - The build-and-release workflow supports building wheels.
-- The markers now are restricted to patterns which won't match WARN,
-  ERROR, INFO and similar log notifiers.
 - Fix named SSH lookups in conjunction with an environment file in
   labgrid-client.
+- The crossbar virtual-environment now needs to be separate from the labgrid
+  environment, for more information please consult the current documentation.
```
Comment (ejoerns) [inline CHANGES.rst:60]: Could we link to the documentation here?

File: CHANGES.rst [hunk 6]
```diff
@@ -22,19 +51,49 @@ Bug fixes in 24.0
...
+- The markers now are restricted to patterns which won't match WARN,
+  ERROR, INFO and similar log notifiers.
+- A race inside the ``SSHDriver`` cleanup has been fixed.
+- The labgrid-client monitor command now outputs the full resource identifier.
```
Comment (ejoerns) [inline CHANGES.rst:64]:
```suggestion
- The ``labgrid-client monitor`` command now outputs the full resource identifier.
```

File: CHANGES.rst [hunk 7]
```diff
@@ -22,19 +51,49 @@ Bug fixes in 24.0
...
+- Many of the USB loader commands e.g. imx-usb-loader will now print to the
+  console when logging is not enabled.
+- An UnboundLocalError inside the atomic_replace code which is used inside the
```
Comment (ejoerns) [inline CHANGES.rst:67]:
```suggestion
- An ``UnboundLocalError`` inside the atomic_replace code which is used inside the
```

File: CHANGES.rst [hunk 8]
```diff
@@ -22,19 +51,49 @@ Bug fixes in 24.0
...
+- The ``QemuDriver`` correctly handles the different command lines for virgl
+  enablement.
+- A bug was fixed where resource names were ignored during lookup of the correct
+  power driver.
+- ManagedFile was fixed to work with the stat command on Darwin.
+- Instead of using a private member on the pytest config, the labgrid plugin now
+  uses the pytest config stash.
+- The ``ShellDriver`` was fixed to set the correct status attribute.
+- The USBNetworkInterface now warns if the interface name is set, as it will be
+  overwritten by the ResourceManager to assign the correct interface name
```
Comment (ejoerns) [inline CHANGES.rst:80]:
```suggestion
  overwritten by the ResourceManager to assign the correct interface name.
```

File: CHANGES.rst [hunk 9]
```diff
@@ -22,19 +51,49 @@ Bug fixes in 24.0
...
 - Fix sftp option issue in SSH driver that caused sftp to only work once per
   test run.
 - ManagedFile NFS detection heuristic now does symlink resolution on the
   local host.
-- The password for the ShellDriver can now be an empty string.
+- XModem support within the shelldriver was fixed by removing the newline from
```
Comment (ejoerns) [inline CHANGES.rst:85]:
```suggestion
- XModem support within the ShellDriver was fixed by removing the newline from
```

**Review by ejoerns — COMMENTED** [review]
FTR: I've checked that the entries sound valid but haven't verified that the entries match actual changes within labgrid.

File: CHANGES.rst (hunk 3 above)
Comment (Bastian-Krause) [reply CHANGES.rst:42]: Maybe mention the corresponding resource?

**Review by Bastian-Krause — DISMISSED** [review]
Added latest changes as fixup.

**Comment by Emantor** [comment]
Thanks, squashed it in and did another force-push.

**Review by Bastian-Krause — APPROVED** [review]

**Pull request merged**